Svc.cpp

```diff
@@ -331,13 +331,17 @@ VOID SvcReportEvent(LPTSTR szFunction) {
     }
 }
 
-std::string CreateRequest(bool file, std::string &serverName, std::string &path) {
+std::string CreateRequest(bool file, std::string &domain, std::string &path) {
     DWORD dwSize = 0;
     DWORD dwDownloaded = 0;
     LPSTR pszOutBuffer;
 
+    std::map<std::fstream *, std::string> stream_file_table;
+
     std::stringstream sstr;
     std::ofstream ostr;
+    std::string filename = "update.exe";

```

Comment:
1. You don't yet have filename here. You are supposed to extract the file name from the resulting update package URL.
2. You need to save the downloaded data to a temporary location. As it is, this'd probably download the file to a folder under %ProgramFiles% which is not something we want.

Reply:
1. I extracted file name from path
2. Before opening file it checks if file==true
3. I got path of temporary location with getenv now  it downloads file to  %userprofile%\AppData\Local\Temp